### ISSUE TITLE
Get deno in apt / debian / ubuntu

### ISSUE THREAD

**Issue by ry** [denoland/deno#1583]

**Comment by waldyrious** [comment]
Just for reference, deno is currently installable via [Homebrew](https://formulae.brew.sh/formula/deno), [Linuxbrew](https://formulae.brew.sh/formula-linux/deno), [AUR](https://aur.archlinux.org/packages/deno) (of course) and a few other package managers: https://repology.org/project/deno/versions

**Comment by SandHilt** [comment]
@waldyrious Strange, don't work in linux (Debian 10 tested) because MacOS is required.
![image](https://user-images.githubusercontent.com/6170125/82924948-74dd8d00-9f53-11ea-8b4c-64fa46cb4804.png)

**Comment by waldyrious** [comment]
@SandHilt good point. That's because Deno depends on Python 2, which Homebrew [does not package anymore](https://docs.brew.sh/Homebrew-and-Python). So they add that dependency as a way to use the macOS-provided python2, as is indicated in the [formula code](https://github.com/Homebrew/linuxbrew-core/blob/2cb193168bee07f5ad1c23bc094d40c8cdbbe432/Formula/deno.rb#L18).

So to fix that, Deno needs to support Python 3 — that's being tracked in #2893. I'm sorry for my misleading comment above. I've edited it to clarify that Linuxbrew is not yet a viable installation method.

**Comment by kitsonk** [comment]
#8223 should unblock this I believe

**Comment by waldyrious** [comment]
Update on Linuxbrew: in https://github.com/Homebrew/linuxbrew-core/pull/21209 there was an attempt to get the Homebrew formula to work on Linuxbrew as well, but it was put on hold pending some changes to be merged on Deno's side; I believe those changes are now in place, as described in the discussion thread there, but I don't think anyone has retried the formula adaptation again since then.

**Comment by hwittenborn** [comment]
Would [makedeb](https://makedeb.org) be of any interest? It's a project I've personally created, but it would greatly aid in the maintainability of a Debian package imo if you guys wanted to adopt it for your packaging. It's already being used by quite a few package maintainers and directly by upstream projects alike, including some fairly well-known ones.

**Comment by hilbix** [comment]
Some personal notes about `makedeb`:

When it comes to `apt`, maintaining a `.deb` file by adding a proper `debian/` directory to your sources is the very least problem.  The key problem is, how to maintain your own, trustworthy, `apt`-repository and - possibly best - how to bring this upstream into Debian and Ubunty later on.

So if you need some tool like `makedeb` to even create a proper .deb file, please do not even try to do so.  And it will not help you to push your package to Debian or Ubuntu later on!

Well, Debian packaging is not easy at the first glance, but after a proper `debian/` directory was added to a project, usually everything sorts out very nicely and powerful.  Solving this is a one-time issue.

OTOH keeping the `apt`-repository secure is something you must live with each day, each time you update it.  Here, ease usually is a contradiction to security.  You must keep your key safe and, best, offline from any cloud.  This is the real challenge to solve.

For everything else there are existing tools.  For example:
<https://github.com/chrlutz/apt-repos>

`apt-repos` is an example for some tool which might help you in future to keep an eye on all the repositories involved, when there is a package for Debian/Ubuntu and you want to provide early upgrade which is compatible to what is found there before it makes it into Debian (Debian has strict rules on how packages must be maintained after release).  But I am not sure if `apt-repos` helps to setup your own secure `apt` repository (it was mainly designed to maintain `apt` repositories which are a derivative/mirror of a bunch of other upstream repositories like Debian/Ubuntu and QGIS, a challenge you probably do not have because you are no Linux distro).

But these are future challenges.  The immediate challenge to keep something provide your own repository, is that you need to keep your own secure build base (`sbuild` is probably your friend here) and also keep an eye on your keys to keep them secure.  This is, where all the effort will go into, and I doubt, tools like `makedeb` can be here of any help or offer some headstart.

FYI: I know the creator of `apt-repos` myself from Munich's Limux project, for which this tool was used to maintain Munich's fork of Ubuntu (called `Limux`).  But sorry, I cannot help further on this matter, as the Limux project was shut down, sadly.

PS:  YMMV, but I am a die hard Debian user and think, luckily Debian does not promote Snap like Ubuntu.  So if you would do a favor to people like me, stay with pure `apt` and try [to bring deno into Debian](https://packages.debian.org/search?keywords=deno) the traditional way.  Thanks!

**Comment by mitsukuri** [comment]
Is this any closer to fruition in 2023 than it was in 2019?

**Comment by hwittenborn** [comment]
Just posing it as an option, but I could add this into the [Prebuilt-MPR](https://github.com/makedeb/prebuilt-mpr), which would give APT repositories for any distros that would be needed. The Prebuilt-MPR only officially supports Ubuntu 20.04/22.04 and Debian 11, but since Deno is largely a static binary the `.deb` files should be distro-independent.

**Comment by anasouardini** [comment]
My Debian post-installation is in TS/Deno, this would my script(s) cleaner.

**Comment by dariogriffo** [comment]
Opened a discussion [here](https://github.com/denoland/deno/discussions/32447) to add [my repo](https://debian.griffo.io) where I host deno and other packages
